src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,48 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Encapsulate Deadline which is-a Task
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
+
+    /**
+     * Class constructor for Deadline.
+     *
+     * @param description Deadline's description.
+     * @param by Deadline's by.
+     * @throws DukeException if Deadline's by is empty.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+        if (by.equals(" ")) {
+            throw new DukeException("☹ OOPS!!! The deadline needs to have a end day and/or date");
+        }
+        try {
+            this.by = LocalDate.parse(by);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("date after /by should be in YYYY-MM-DD Format.");
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString()
+                + " (by: " + this.by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
+
+    /**
+     * Class constructor for Deadline.
+     *
+     * @return string that represent this deadline object, to be stored in data file.
+     */
+    @Override
+    public String toStorageString() {
```

Comment:
I like your use of inheritance in the toStorageString method

Reply:
Thanks.